File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.cpp
class Solution {
public:
    string sortVowels(string s) {
        unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        int n = s.length();
        priority_queue<char, vector<char>, greater<char>> q; 
        for(int i = 0; i < n; i++)
        {
            if(vowels.find(s[i]) != vowels.end())
            {
                q.push(s[i]);
            }
        }
        for(int i = 0; i < n; i++)
        {
            if(vowels.find(s[i]) != vowels.end())
            {
                s[i] = q.top();
                q.pop();
            }
        }
        return s;
    }
};

File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = set('aeiouAEIOU')

        charVowels = [ch for ch in s if ch in vowels]
        charVowels.sort()

        result = []
        vowelIndex = 0

        for ch in s:
            if ch in vowels:
                result.append(charVowels[vowelIndex])
                vowelIndex += 1
            else:
                result.append(ch)
        
        return ''.join(result)
